Update README to include the custom exception handler (#22)
errm/promenade: README.md

The exporter runs by default on port `9394` and the metrics are available at the standard path of `/metrics`, the stand-alone exporter is configured to use gzip.

> ### Rack Middleware

Reviewer: nit: Jus thinking now, would it be more explicit to call it Rails Middleware in the doc? what about the class name?

```suggestion
### Rails Middleware
```

Promenade provides custom Rack middleware to track HTTP response times for requests in your Rack application.

This was originally inspired by [prometheus-client-mmap](https://gitlab.com/gitlab-org/prometheus-client-mmap/-/blob/master/lib/prometheus/client/rack/collector.rb).

> **This middleware is automatically added to your Rack stack if your application is a Ruby on Rails app.**

Reviewer: ```suggestion
**This middleware is automatically added to your [middleware stack](https://guides.rubyonrails.org/rails_on_rack.html#inspecting-middleware-stack)**
```

Author: I think we want to make it clear that this only happens when the app is a Rails app though?

Reviewer: correct

> We recommend you add the middleware after `ActionDispatch::ShowExceptions` in your stack, so you can accurately record the controller and action where an exception was raised.
> 
> If you want to change the position, or customise the labels and exception handling behaviour, simply remove the middleware from the stack and re-insert it with your own preferences.
> 
> ``` ruby
> Rails.application.middleware.delete(Promenade::Client::Rack::Collector)
> Rails.application.middleware.insert_after(Rails::Rack::Logger, Promenade::Client::Rack::Collector)
> ```
> 
> #### Customising the labels recorded for each request
> 
> If you would like to collect different labels with each request, you may do so by customising the middleware installation:
> 
> ``` ruby
> label_builder = Proc.new do |env|
>   {
>     method: env["REQUEST_METHOD"].to_s.downcase,
>     host: env["HTTP_HOST"].to_s,
>     controller: env.dig("action_dispatch.request.parameters", "controller") || "unknown",
>     action: env.dig("action_dispatch.request.parameters", "action") || "unknown"
>   }
> end
> Rails.application.config.middleware.insert_after ActionDispatch::ShowExceptions,
>         Promenade::Client::Rack::Collector
>         label_builder: label_builder
> ```
> 
> #### Customising how the middleware handles exceptions
> 
> The default implementation will capture exceptions, count the execption class name (e.g. `"StandardError"`), and then re-raise the exception.
> 
> If you would like to customise this behaviour, you may do so by customising the middleware installation:
> 
> ``` ruby
> exception_handler = Proc.new do |exception, exception_counter, env_hash, request_duration_seconds|
>   # This simple example just re-raises the execption
>   raise exception
> end
> Rails.application.config.middleware.insert_after ActionDispatch::ShowExceptions,
>         Promenade::Client::Rack::Collector
>         exception_handler: exception_handler
> ```
> 

Reviewer: I think we don't need these instructions anymore with the including of our Railtie class WDYT?
```suggestion
```

Author: These are for people who want to configure away from the default behaviour. I think it's important to keep it, otherwise how would they know it's configurable? 🤷

Reviewer: Yeah dunno why I thought we removed these options, make sense now 👍
